converter.go

```diff
@@ -361,11 +391,112 @@ func stringToValue(dest *driver.Value, srcColumnMeta execResponseRowType, srcVal
 		}
 		*dest = b
 		return nil
+	case "array":
+		if len(srcColumnMeta.Fields) == 0 {
+			*dest = *srcValue
+			return nil
+		}
+		if len(srcColumnMeta.Fields) > 1 {
+			return errors.New("got more than one field for array")
+		}
+		v, err := buildStructuredArray(srcColumnMeta.Fields[0], srcValue, params)
+		if err != nil {
+			return err
+		}
+		*dest = v
+		return nil
 	}
 	*dest = *srcValue
 	return nil
 }
 
+func buildStructuredArray(fieldMetadata fieldMetadata, srcValue *string, params map[string]*string) (any, error) {
```

Comment:
This is for JSON or Arrow format with JSON structured types, correct?

Reply:
Correct.